sdk/core/Azure.Core/src/Shared/AutoRest/ChangeTrackingList.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+#nullable enable
+
+namespace Azure.Core
+{
+    internal class ChangeTrackingList<T>: IList<T>, IReadOnlyList<T>
```

Comment:
How doe this class track changes to the list?

Reply:
It tracks if the list was modified/initialized and needs to be serialized.

By default non-required collections are not serialized until customer modifies them or they were defined when received from the service.

Reply:
Thanks for the offline chat - I misunderstood this as tracking changes to the list post-initialization, however it is intended to track whether the value was defined or not.